rest_framework/serializers.py

```diff
@@ -985,13 +992,16 @@ def restore_object(self, attrs, instance=None):
             if field_name in attrs:
                 m2m_data[field_name] = attrs.pop(field_name)
 
-        # Forward m2m relations
-        for field in meta.many_to_many + meta.virtual_fields:
+        def _inner_loop_code(field):
```

Comment:
I'm confused about this - what's going on here, and should this really be part of the same pull request as the SortedDict/OrderedDict change?